volatility3/framework/plugins/windows/pe_symbols.py

```diff
@@ -0,0 +1,777 @@
+# This file is Copyright 2024 Volatility Foundation and licensed under the Volatility Software License 1.0
+# which is available at https://www.volatilityfoundation.org/license/vsl-v1.0
+
+
+import io
+import logging
+
+from typing import Dict, Tuple, Optional, List, Generator, Union
+
+import pefile
+
+from volatility3.framework import interfaces, exceptions
+from volatility3.framework import renderers, constants
+from volatility3.framework.configuration import requirements
+from volatility3.framework.renderers import format_hints
+from volatility3.framework.symbols import intermed
+from volatility3.framework.symbols.windows import pdbutil
+from volatility3.framework.symbols.windows.extensions import pe
+from volatility3.plugins.windows import pslist, modules
+
+vollog = logging.getLogger(__name__)
+
+
+class PESymbolFinder:
+    """
+    Interface for PE symbol finding classes
+    This interface provides a standard way for the calling code to
+    lookup symbols by name or address
+    """
+
+    cached_str = Union[str, None]
```

Comment:
`out in the module proper` means outside of the class? Just want to verify.

Reply:
Yeah, outside of the class, but within this module.  So you'd do `pe_symbols.cached_str`  rather than `PeSymbols.cached_str`.  I'm just after them being tidier and better named somehow?  I'm even ok duplicating these where they're used and then letting the type checker make sure everything's in the right place?

Reply:
Yea thats good. I am moving them out now.